google/cloud/sql/connector/InstanceConnectionManager.py

```diff
@@ -439,19 +403,56 @@ def immediate_future(object: Any) -> concurrent.futures.Future:
         fut.set_result(object)
         return fut
 
-    def connect(self,) -> OpenSSL.SSL.Connection:
+    def connect(self, driver: str, username:str=None, **kwargs) -> OpenSSL.SSL.Connection:
         """A method that returns an OpenSSL connection to the database.
 
         :rtype: OpenSSl.SSL.Connection
         :returns: An OpenSSL connection to the primary IP of the database.
         """
+        logger.debug("Entered connect method")
+        if username is None:
+            raise ValueError("No username provided!")
+
         with self._lock:
             instance_data: InstanceMetadata = self._current.result()
-        ctx = instance_data.ssl_context
-        ip_addr = instance_data.ip_addresses["PRIMARY"]
-        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        ssl_connection = OpenSSL.SSL.Connection(ctx, s)
-        ssl_connection.connect((ip_addr, 3307))
-        ssl_connection.do_handshake()
-
-        return ssl_connection
+        
+        ssl_ca = NamedTemporaryFile(suffix=".pem")
+        ssl_cert = NamedTemporaryFile(suffix=".pem")
+        ssl_key = NamedTemporaryFile(suffix=".pem")
+
+        ssl_ca.write(instance_data.server_ca_cert.encode())
+        ssl_cert.write(instance_data.ephemeral_cert.encode())
+        ssl_key.write(instance_data.private_key)
+
+        if driver is "pg8000":
+            try:
+                import pg8000
+                logger.debug("Successfully imported %s", driver)
+
+                ssl_dict = {
+                    "ca_certs": ssl_ca.name,
+                    "certfile": ssl_cert.name,
+                    "keyfile": ssl_key.name
+                }
+                logger.debug("Temporary files: {}, {} and {}".format(ssl_ca.name, ssl_cert.name, ssl_key.name))
+                
+                return pg8000.connect(username, instance_data.ip_addresses["PRIMARY"], ssl=ssl_dict, **kwargs)
+            except ImportError as e:
+                raise e
```

Comment:
You shouldn't catch an exception if you are just going to throw it again unchanged (it's an antipattern). You should show a message explaining to the use that they need to have the driver installed if they want to use it. 

Additionally, the scope of this try/except is too big - as a general rule, you should limit to code that throws the error to avoid catching something else.